R/utils.R

```diff
@@ -0,0 +1,28 @@
+# sgdnet: Penalized Generalized Linear Models with Stochastic Gradient Descent
+# Copyright (C) 2018  Johan Larsson
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#' Retrieve Response Family from sgdnet Object
+#'
+#' @param x an object of class `'sgdnet'`
+#'
+#' @return The family that was used in fitting `x`.
+#'
+#' @keywords internal
+extract_family <- function(x) {
+  supported_families <- c("gaussian", "binomial")
+  family_index <- inherits(x, supported_families, TRUE) > 0
+  supported_families[family_index]
+}
```

Comment:
I don't like storing family information in a S3 class tag like this. `sgdnet` is not a subclass of `gaussian` objects (since there is no "gaussian") object. Is there any reason not to just store it as a regular part of an S3 object? If we need it for S3 dispatch somewhere, I'd much rather have types like `c('gaussian_sgdnet', 'sgdnet')` since that's a proper subtype.